app/services/content_sanitizer.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# Sanitizes and cleans up user-provided content that may contain tags, special characters, etc.
+
+class ContentSanitizer
+  include ActionView::Helpers::SanitizeHelper
+
+  ALLOWED_TAGS = ["p", "b", "strong", "em", "i", "a", "u"].freeze
+  ALLOWED_ATTRIBUTES = ["href", "target"].freeze
+  FILTERED_CHARACTERS = {
+    "&amp;amp;" => "&",
+    "&amp;" => "&",
+    "&nbsp;" => " "
+  }.freeze
+
+  def strip_content(content)
+    content = strip_tags(content.to_s.strip)
+
+    filter_characters(content) if content.length
+  end
+
+  def sanitize_content(content)
+    content = sanitize(content.to_s, tags: ALLOWED_TAGS, attributes: ALLOWED_ATTRIBUTES)
+
+    filter_characters(content) if content.length
```

Comment:
0 is truthy so I'm afraid this will still filter if there's no content.

```
irb(main):004:0> puts 'foo' if "".length
foo
=> nil
```

Reply:
5723a79409ecedab83fe9f3c1c1d051d1129239a 👍